extensions/sql-database-projects/src/dialogs/deployDatabaseDialog.ts

```diff
@@ -0,0 +1,291 @@
+
+		this.dialog.cancelButton.label = constants.cancelButtonText;
+		this.dialog.cancelButton.onClick(async () => await this.cancel());
+		this.dialog.cancelButton.onClick(async () => await this.cancel());
+
+		let generateScriptButton: azdata.window.Button = azdata.window.createButton(constants.generateScriptButtonText);
+		generateScriptButton.onClick(async () => await this.generateScript());
+		generateScriptButton.enabled = false;
+
+		this.dialog.customButtons = [];
+		this.dialog.customButtons.push(generateScriptButton);
+
+		azdata.window.openDialog(this.dialog);
+	}
+
+
+	private initializeDialog(): void {
+		this.initializeDeployTab();
+		this.dialog.content = [this.deployTab];
+	}
+
+	private initializeDeployTab(): void {
+		this.deployTab.registerContent(async view => {
+
+			let selectConnectionRadioButtons = this.createRadioButtons(view);
+			this.targetConnectionFormComponent = this.createTargetConnectionComponent(view);
+
+			this.targetDatabaseTextBox = view.modelBuilder.inputBox().withProperties({
+				value: this.getDefaultDatabaseName(),
+				ariaLabel: constants.databaseNameLabel
+			}).component();
+
+			this.dataSourcesFormComponent = this.createDataSourcesDropdown(view);
+
+			this.targetDatabaseTextBox.onTextChanged(() => {
+				this.tryEnableGenerateScriptAndOkButtons();
+			});
+
+			this.deployScriptNameTextBox = view.modelBuilder.inputBox().withProperties({
+				value: this.getDefaultScriptName(),
+				ariaLabel: constants.deployScriptNameLabel
+			}).component();
+
+			this.deployScriptNameTextBox.onTextChanged(() => {
+				this.tryEnableGenerateScriptAndOkButtons();
+			});
+
+			this.formBuilder = <azdata.FormBuilder>view.modelBuilder.formContainer()
+				.withFormItems([
+					{
+						title: constants.targetDatabaseSettings,
+						components: [
+							{
+								title: constants.selectConnectionRadioButtonsTitle,
+								component: selectConnectionRadioButtons
+							},
+							this.targetConnectionFormComponent,
+							{
+								title: constants.databaseNameLabel,
+								component: this.targetDatabaseTextBox
+							},
+							{
+								title: constants.deployScriptNameLabel,
+								component: this.deployScriptNameTextBox
+							}
+						]
+					}
+				], {
+					horizontal: false
+				})
+				.withLayout({
+					width: '100%'
+				});
+
+			let formModel = this.formBuilder.component();
+			await view.initializeModel(formModel);
+		});
+	}
+
+	private async deploy(): Promise<void> {
+		// TODO: hook up with build and deploy
+		// if target connection is a data source, have to check if already connected or if connection dialog needs to be opened
+	}
+
+	private async generateScript(): Promise<void> {
+		// TODO: hook up with build and generate script
+		// if target connection is a data source, have to check if already connected or if connection dialog needs to be opened
+		azdata.window.closeDialog(this.dialog);
+	}
+
+	private async cancel(): Promise<void> {
```

Comment:
I think it makes sense to leave this as cancel since this doesn't close the dialog since the dialog already handles closing the dialog when cancelButton is clicked. This also doesn't do anything, so it can also just be removed for now.